### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by AlbertGaz**

### CODE REVIEW

File: hw/hw2/task1.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    Returns: max_words. list of 10 longest words
+
+    """
+    with open(file_path, encoding="utf-8") as f:
+        max_words = ["" for _ in range(10)]
+        for line in f:
+            for word in line.split():
+                for i in range(10):
+                    if len(set(word)) > len(set(max_words[i])):
```
Comment (Aleksei-Samarin) [inline hw/hw2/task1.py]: Too many nested code here. Can we use defaultdict/Counter to simplify it?

File: hw/hw2/task1.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    Args:
+        file_path: text file direction
+
+    Returns: rarest char
+
+    """
+    with open(file_path, encoding="utf-8") as f:
+        text = "".join(f.read().split())  # get rid of white space in text string
+        text_chars = set(text)
```
Comment (Aleksei-Samarin) [inline hw/hw2/task1.py]: We can delete one level of nested code like this here and below in other tasks:
```
with open(file_path, encoding="utf-8") as f:
    text = "".join(f.read().split())  # get rid of white space in text string
text_chars = set(text)
...
```

File: hw/hw2/task3.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    Args:
+        *args: list of elements
+
+    Returns: list of combination lists
+
+    """
+    res = []
+    for i in product(*args):
+        res.append(list(i))
```
Comment (Aleksei-Samarin) [inline hw/hw2/task3.py]: Great solution, and now we can transform it to one-liner

File: hw/hw2/task5.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    Returns: list of objects iterated
+
+    """
+    if len(arg) == 3:
+        start = arg[0]
+        end = arg[1]
+        step = arg[2]
+    elif len(arg) == 2:
+        start = arg[0]
```
Comment (Aleksei-Samarin) [inline hw/hw2/task5.py]: These three conditions can be simplified, for example, for second block: 
`start, end, step = *arg, 1`

**Review by Aleksei-Samarin — CHANGES_REQUESTED** [review]

File: hw/hw2/task1.py (hunk 1 above)
Comment (AlbertGaz) [reply hw/hw2/task1.py]: Come up with a better solution using sorted by key len(set(x))

**Pull request merged**